### PR TITLE
Support esm for module loading using option flag

### PR DESCRIPTION
**Pull request by rtablada**
This PR adds support to use [ESM](https://github.com/standard-things/esm) to load ES6 modules as mentioned in #651 and would possibly be a resolution for a lot of people in #112 who's main concern is loading modules and not full transpilation.

I tested on my codebase and this works for `db:create`, `db:drop`.

~However further updates to `umzug` will be needed to support this for to work for seeds and migrations.~

### CODE REVIEW

**Comment by rtablada** [comment]
Because of ESM support engines this requires node version >6.

Because of node support as mentioned in #655 this should be the support stack going forward.

**Comment by rtablada** [comment]
After looking through `umzug` source, by passing a `customResolver` function using ESM, then ESM modules are supported and work for all migration and seed commands.

File: src/commands/migrate.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ function migrationStatus(args) {
   }).catch(e => helpers.view.error(e));
 }
 
-function migrateSchemaTimestampAdd(args) {
+function migrateSchemaTimestampAdd (args) {
```
Comment (ebryn) [inline src/commands/migrate.js]: @rtablada you want to remove these unnecessary whitespace changes?

Comment (rtablada) [reply src/commands/migrate.js]: The changes seem to be from the eslint config in the project that was missing. I can try to turn off autoformat/fix in my config.

**Comment by sushantdhiman** [comment]
I am reluctant to add this change as it doesn't actually introduce anything of significant use apart from ES6/7 style `import`.

Please check https://github.com/sequelize/cli/issues/112#issuecomment-394172092 / [docs](https://github.com/sequelize/sequelize/blob/master/docs/migrations.md#using-babel) to see how you can use `babel` with `sequelize-cli` setup, which will allow you to use full ES6/7 setup without much friction.

I may include this change in future CLI v5, but chances are slim.

Thanks for contributing to CLI. If my solution doesn't work for you, please let me know.

**Comment by ebryn** [comment]
> it doesn't actually introduce anything of significant use apart from ES6/7 style import.

@sushantdhiman that's the whole point. you don't consider ES modules significantly useful?

**Comment by ebryn** [comment]
@sushantdhiman i believe it's also fair to say that you should be recommending using `@std/esm` over `babel` nowadays. maybe @jdalton can chime in?

**Comment by jdalton** [comment]
Hi @ebryn!

Instead of baking in `esm` support you could add a `--require` (`-r` alias) this would allow `esm`, `@babel/register`, `ts/register`, and others to be used.

**Comment by sonufrienko** [comment]
I use ESM to load ES Modules instead of Babel.
So this is a wanted feature.

Thanks, @rtablada  :+1:
